### DISCUSSION TITLE
Regarding memory allotment and time for computation

### DISCUSSION THREAD

**Discussion by kumaryash7** [pybamm-team/PyBaMM#3788] [Q&A]
I am running an optimization algo in which I am trying to minimize degradation and time of charging. I am running for 500 cycles but in each optimization iteration it runs for 500 cycles and it takes too much time and memory. 
Even if I have enough memory, it takes too much of time to run. 
If anyone has any idea about this, please let me know.
Thank you

**Comment by TomTranter** [comment]
You can save memory by only storing some solutions as explained here https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/examples/notebooks/simulations_and_experiments/simulating-long-experiments.ipynb
There's no easy answer on how to speed things up. One method is to try figuring out what the LAM and LLI is by fitting these parameters or associated ones directly rather than running every cycle with degradation submodels turned on.

**Reply by kumaryash7** [reply]
Thank you for the reply.
Basically, I am trying to minimize degradation and charging time simultaneously using genetic algo. So for the algo to run, it needs to solve for 500 cycles in each iteration, therefore it is taking too much time.
I can try out the example for the memory problem.
Also can you elaborate on fitting LAM and LLI parameters?

**Reply by TomTranter** [reply]
I had assumed you had some data you were trying to fit to in which case you can manually adjust the amount of active material and lithium balance without running the degradation models but if you are simulating degradation there isn't much you can do to speed up except maybe you can look at running intra-cycle updates like this paper https://iopscience.iop.org/article/10.1149/1945-7111/ad1294/meta
